rtl/eclrtl/rtldynfield.cpp

```diff
@@ -1147,7 +1147,7 @@ class GeneralRecordTranslator : public CInterfaceOf<IDynamicTransform>
         bool hasBlobs = false;
         if (!estimate)
         {
-            if (binarySource)
+            if (binarySource && !hasBlobs)
```

Comment:
Can hasBlobs be true here ?

Reply:
No, that's the result of an incorrect paste in the first version of this fix that I had forgotten was not part of the original code!

I have removed, squashed, and repushed